### PR TITLE
ListOfMatches date search, ListOfStudents activity column, Register page bugfix

### PR DESCRIPTION
**Pull request by EAkkanen**
1) ListOfMatches admin page date range search feature
Many changes required for this one. Three new DOM elements were added: 2 datepickers and a "reset dates" button. Various new methods were needed to support date range filtering (handleToDateChange, handleFromDateChange, compareDateRange, onResetDates). The "handleSearchChange" method was replaced with "handleAllSearchChanges" that filters the results twice, based on the textfield search criteria as well as the date range criteria.

2) ListOfStudents admin page shows student activity
The column for user activity now correctly displays the user's activity status with either a check icon or a red X (called a "close icon" in MUI).

3) Register page/Sign up page bugfix
A simple fix for a small bug. Previously it was possible for a new user to choose the same language multiple times for "want to learn".

### CODE REVIEW

**Review by robin-papp — APPROVED** [review]

File: src/containers/ListOfMatches/index.js [hunk 1]
```diff
@@ -104,22 +115,95 @@ class ListOfMatches extends React.Component
...
-    if (event.target.value.length >= 2){
-      let searchResult = this.state.data.filter(item => {
-        return item.requesterUser.lastName.toLowerCase().includes( searchValue) || item.recipientUser.lastName.toLowerCase().includes( searchValue)
-        ||item.requesterUser.firstName.toLowerCase().includes( searchValue) || item.recipientUser.firstName.toLowerCase().includes( searchValue)
-        ||item.requesterUser.email.toLowerCase().includes( searchValue) || item.recipientUser.email.toLowerCase().includes( searchValue);
 
+  textfieldChange = (event) =>{
+    //This is to fix asynchronization issue
+    this.setState({searchValue: event.target.value}, function(){
+      console.log("searchValue's setState completed", this.state);
```
Comment (janakavishwajith) [inline src/containers/ListOfMatches/index.js]: Please remove the console.logs, it doesn't looks nice on production.. :P

File: src/containers/ListOfMatches/index.js [hunk 2]
```diff
@@ -104,22 +115,95 @@ class ListOfMatches extends React.Component
...
     }
   }
+
+  //Check whether a date is included in the specified range
+  //Used in filtering the table according to date search criteria
+  compareDateRange = (dateval) =>{
+    let targetdate = parseISO(dateval).setHours(0, 0, 0, 0);;
+    let date1 = this.state.fromDateValue.setHours(0, 0, 0, 0);
+    let date2 = this.state.toDateValue.setHours(0, 0, 0, 0);
+    if(isAfter(targetdate, date1) || isEqual(targetdate, date1)){
```
Comment (janakavishwajith) [inline src/containers/ListOfMatches/index.js:156]: This logic could be more simplified...
Not a mandatory change

File: src/containers/ListOfMatches/index.js [hunk 3]
```diff
@@ -104,22 +115,95 @@ class ListOfMatches extends React.Component
...
+      return false;
+    }
+    else{
+      return false;
+    }
+  }
+
+  handleFromDateChange = (date) => {
+    this.setState({fromDateValue:date}, function(){
+      console.log("handleFromDateChange's setState completed", this.state);
```
Comment (janakavishwajith) [inline src/containers/ListOfMatches/index.js]: Please remove the console logs

File: src/containers/ListOfMatches/index.js [hunk 4]
```diff
@@ -104,22 +115,95 @@ class ListOfMatches extends React.Component
...
+      else{
+        this.handleAllSearchChanges();
+      }
+    });
+  }
+
+  handleToDateChange = (date) => {
+    //Fixing another asynchronization issue
+    this.setState({toDateValue:date}, function(){
+      console.log("handleToDateChange's setState completed", this.state);
```
Comment (janakavishwajith) [inline src/containers/ListOfMatches/index.js]: More logs to remove

**Review by janakavishwajith — CHANGES_REQUESTED** [review]
Some smaller changes to be done....

**Comment by EAkkanen** [comment]
Made Janaka's requested changes of removing console logs for asynchronization checks.
Added a maxDate value to "to-date" as well that limits the latest selectable date to "current date".

**Review by janakavishwajith — APPROVED** [review]

**Pull request merged**